src/main/java/com/pinterest/secor/common/KafkaClient.java

```diff
@@ -212,6 +219,16 @@ public Message getCommittedMessage(TopicPartition topicPartition) throws Excepti
             }
             consumer = createConsumer(topicPartition);
             return getMessage(topicPartition, committedOffset, consumer);
+        } catch (MessageDoesNotExistException e) {
+          // If a RuntimeEMessageDoesNotExistException exception is raised,
+          // the message at the last comitted offset does not exist in Kafka.
+          // This is usually due to the message being compacted away by the
+          // Kafka log compaction process.
+          //
+          // That is no an exceptional situation - in fact it can be normal if
+          // the topic being consumed by Secor has a low volume. So in that
+          // case, simply return null
+          return null;
```

Comment:
I am fine with either way.  The caller either needs to deal with that special exception or check null return value.